packages/core/src/factory.ts

```diff
@@ -1,65 +1,68 @@
-import { DiagnosticSeverity } from '@stoplight/types';
 import * as Either from 'fp-ts/lib/Either';
 import { getOrElse, map } from 'fp-ts/lib/Option';
 import { pipe } from 'fp-ts/lib/pipeable';
 import * as TaskEither from 'fp-ts/lib/TaskEither';
 import { defaults } from 'lodash';
-import { IPrism, IPrismComponents, IPrismConfig, IPrismDiagnostic, ProblemJsonError } from './types';
+import { IPrism, IPrismComponents, IPrismConfig, IPrismDiagnostic, IPrismProxyConfig } from './types';
 import { validateSecurity } from './utils/security';
 
+function isProxyConfig(p: IPrismConfig): p is IPrismProxyConfig {
+  return !p.mock;
+}
+
 export function factory<Resource, Input, Output, Config extends IPrismConfig>(
   defaultConfig: Config,
-  components: IPrismComponents<Resource, Input, Output, Config>,
+  components: IPrismComponents<Resource, Input, Output, Config>
 ): IPrism<Resource, Input, Output, Config> {
   return {
-    request: async (input: Input, resources: Resource[], c?: Config) => {
+    request: (input: Input, resources: Resource[], c?: Config) => {
       // build the config for this request
       const config = defaults<unknown, Config>(c, defaultConfig);
       const inputValidations: IPrismDiagnostic[] = [];
 
       return pipe(
         TaskEither.fromEither(components.route({ resources, input })),
         TaskEither.chain(resource => {
-          if (config.validateRequest && resource) {
+          if (config.validateRequest) {
             inputValidations.push(
               ...components.validateInput({
                 resource,
                 element: input,
-              }),
+              })
             );
           }
 
-          const inputValidationResult = config.checkSecurity
-            ? inputValidations.concat(
-                pipe(
+          config.checkSecurity
+            ? inputValidations.push(
```

Comment:
so the mutation was needed so that https://github.com/stoplightio/prism/pull/669/files#diff-47c5dc2d65fd624c869f5f08d0cfb56aL73 have it or there was some other reason?

Reply:
Yeah; do not worry about this so much because the validation refactor #731 will remove all this bad stuff.